### DISCUSSION TITLE
Community Mods Market?

### DISCUSSION THREAD

**Discussion by LikiaBili** [ppy/osu#26987] [Ideas] [answered]
I have a **idea** that since theres a lot of features can't be implemented in to the game, then why don't make a community mods market?
Like there can be Ranked mods and Unranked mods which give pp or not, such as a SS and Perfect Indicator can be Ranked (x1.0), and selecting a spicifc point to start playing in order to practice can be Unranked
and you can install community mods as easy as just downloading beatmaps

**Comment by rodrigopina360** [comment]
I am guessing that could be a possibility, game already supports rulesets, I don't see why a community made "unverified" mods tab couldn't be a thing. I personally like the idea, but there's probably a million other things more urgent than that currently

**Reply by LikiaBili** [reply]
Yeah, so this might be a thing after quite some time, a year? three years? understandable

**Comment by exformation** [comment]
From the discussion at <https://github.com/ppy/osu/pull/27078> I think this would probably be the best solution.

Players can gain access to entirely different rhythm games easily through dropping files in `osu/rulesets`. How about allowing them to do the same through `osu/mods`?

The core lazer mods can be stabilized and anyone wanting something more niche can be pointed towards making their own custom mods which are easily importable into the main client. 

This would allow for a community of mod making to spring up like has happened with ruleset making. 

I have at least a dozen ideas for mods, and I'm sure others do as well, so this sounds like it would be a good solution to that. I'll start looking into how to do this, but any pointers would be appreciated.

**Reply by peppy** [reply]
This would not be supported by the core game. It is not on our roadmap.

**Reply by exformation** [reply]
Could this change in the future? If people wanted to make their own mods for core rulesets, should they try to PR them into lazer? I'd like to make mods for mania/taiko, so if a 3rd party system can't be implemented, and merging them with core isn't easy, then I imagine anyone wanting custom mods would just have to play on release builds of their own lazer forks.

**Reply by exformation** [reply]
or I guess I could make rulesets that are copy-pastes of taiko/mania and add mods to those. Then I can just drop those rulesets into `osu/rulesets` and play a version of taiko/mania that has my mods but on a stable build of lazer.

**Reply by peppy** [reply, marked as answer]
> Could this change in the future?

Maybe years in the future. The amount of work required to make this happen is non-trivial. And I am not about to support users loading tens or hundreds of wild dlls (ie. if a user had a mod loaded, all online play would need to be disabled).

Thus it's not something we'd be looking at for now.

**Comment by Rekunan** [comment]
> selecting a spicifc point to start playing in order to practice can be Unranked

This is already in the long term plans, you can see this by searching through the discussions (notable one: https://github.com/ppy/osu/discussions/13013#discussioncomment-2467752).
